lms/djangoapps/discussion/rest_api/tests/test_api.py

```diff
@@ -2657,6 +2684,35 @@ def test_invalid_field(self):
             update_thread(self.request, "test_thread", {"raw_body": ""})
         assert assertion.value.message_dict == {'raw_body': ['This field may not be blank.']}
 
+    @ddt.data(
+        FORUM_ROLE_ADMINISTRATOR,
+        FORUM_ROLE_MODERATOR,
+        FORUM_ROLE_COMMUNITY_TA,
+        FORUM_ROLE_STUDENT,
+    )
+    def test_update_thread_with_edit_reason_code(self, role_name):
+        """
+        Test editing comments, specifying and retrieving edit reason codes.
+        """
+        _assign_role_to_user(user=self.user, course_id=self.course.id, role=role_name)
+        self.register_thread()
+        try:
+            result = update_thread(self.request, "test_thread", {
```

Comment:
@xitij2000 
I've been unable to get `httpretty` mock to work with these calls, for some reason, so these tests are still failing.
Do you know what might be missing here?

Reply:
I've looked into this a bit, and it seems I have figured out a reason. The issue here is in your modifications to `_get_thread_callback`  and `_get_comment_callback`. If the callback function throws an error, then it seems that the mock is not used. 

Right now there are two sources of errors here:
1. There is no `raw_body` key, it should be `body`. 
2. There is no request.user.username. I think the mock request doesn't have that info. This is a regular request, not a django request so I think you'll need to figure out some other way for the username, or simply not assert on that.

Reply:
Thanks, that was the reason, the username should have come from the thread/comment data instead.
Now the tests are passing.